source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/ServiceProxy.cs

```diff
@@ -15,41 +16,16 @@ protected void SetTargetService(object service)
         }
 
         public override object Invoke(MethodInfo method, object[] args)
-        {
-            EnsureTargetServiceNotNull();
-            try
-            {
-                OnStartingInvocation(method);
-
-                return method.Invoke(TargetService, args)!;
-            }
-            catch (TargetInvocationException e)
-            {
-                OnInvocationException(method, e.InnerException!);
-
-                //we need to unwrap the TargetInvocationException
-                ExceptionDispatchInfo.Capture(e.InnerException).Throw();
-
-                //this will never be hit because the line above will throw an exception
-                return default;
-            }
-            catch (Exception e)
-            {
-                OnInvocationException(method, e);
-                throw;
-            }
-            finally
-            {
-                OnCompletingInvocation(method);
-            }
-        }
+            => throw new InvalidOperationException("We do not support decorating synchronous service methods.");
```

Comment:
We no longer have synchronous services, so we can just thrown an exception if it tries to execute